### ISSUE TITLE
[feature request] Allow requesting spot instance instead of an on-demand instance.

### ISSUE THREAD

**Issue by ohinds** [machulav/ec2-github-runner#169]
It would be nice to be able to request a spot instance when it fits the use-case, to reduce cost. 

I think it would be straightforward to add a conditional in `startEc2Instance` (or another function entirely) to call `requestSpotInstances` instead of `runInstances` [here](https://github.com/machulav/ec2-github-runner/blob/2c4d1dcf2c54673ed3bfd194c4b6919ed396a209/src/aws.js#L53).

**Comment by Preen** [comment]
+1

**Comment by troinine** [comment]
This would be a great addition. The current AWS SDK for Javascript v2 `EC2.runInstances()` allows definition of market options which would probably serve a good amount of use cases.

`requestSpotInstances` is deprecated and should not be used as per https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-best-practices.html#which-spot-request-method-to-use

**Comment by tknisch** [comment]
+1 for this!

**Comment by troinine** [comment]
Proposing to address this with #171. I have conducted few tests with an actual AWS account.

**Comment by Preen** [comment]
If I look at the pull requests, Im a bit worried that this repo isnt maintained anymore?

**Comment by Preen** [comment]
@tonyhutter u were the last to commit changes to this Repo. Maybe you can take a look at the proposed solution?

**Comment by Dmitry1987** [comment]
> If I look at the pull requests, Im a bit worried that this repo isnt maintained anymore?

the maintainer is looking for others who can take over the support and maintenance https://github.com/machulav/ec2-github-runner/discussions/172  i proposed to help, will do my best within limited scope, but we need more people 💯 . let's see what the decision will be. not many chime in so far though 😓 because nobody sees the announcement, it should be inserted to the readme header.

**Comment by ykostsik** [comment]
Hi all, 
Just wondering if these changes are going to be merged in the v2?

**Comment by Preen** [comment]
@ykostsik its already merged. Use the latest v2.x

**Comment by ykostsik** [comment]
@Preen Yeah I see it's available in v2.3.8. I just wanted to ask if there are any plans to backport this feature into the v2 tag?
https://github.com/machulav/ec2-github-runner/compare/v2...v2.3.8 because many people just use this as described in the docs
` uses: machulav/ec2-github-runner@v2`
thanks
